src/solver/ts-generator/thermal.cpp

```diff
@@ -575,7 +559,7 @@ void GeneratorTempData::operator()(Data::Area& area, Data::ThermalCluster& clust
                 double AVPDayInTheYear = AVP[dayInTheYear];
                 for (uint h = 0; h != 24; ++h)
                 {
-                    dstSeries[hour] = std::round(AVPDayInTheYear * modulation[hour]);
+                    dstSeries[hour] = std::round(AVPDayInTheYear * cluster.modulationCapacity[hour]);
```

Comment:
You can keep the modulation reference, and just change its target (`cluster.modulationCapacity`)